src/ArrayValueCalculator.java

```diff
@@ -0,0 +1,33 @@
+public class ArrayValueCalculator {
+
+    public static String doCalc(String stringArray[][]) throws ArraySizeException, ArrayDataException {
+        int sum = 0;
+
+        for (int i = 0; i < stringArray.length; i++) {
+            for (int j = 0; j < stringArray[i].length; j++) {
+                if (i <= 0 || j <= 0) {
+                    throw new IllegalArgumentException("Each element in the array must be non-negative and greater than zero");
+                }
+            }
+        }
+
+            if (stringArray == null) {
+                throw new NullPointerException("The array must not be null, the array must be 4x4");
+            }
+            if (stringArray.length != 4 || stringArray[0].length != 4) {
+                throw new ArraySizeException("The array must be 4x4");
+            }
+
+            for (int i = 0; i < stringArray.length; i++) {
+                for (int j = 0; j < stringArray[i].length; j++) {
+                    try {
+                        int number = Integer.parseInt(stringArray[i][j]);
```

Comment:
Simplify this `int number = Integer.parseInt(stringArray[i][j]); sum += number;`
to `sum += Integer.parseInt(stringArray[i][j]);`